### DISCUSSION TITLE
Socket Exception error

### DISCUSSION THREAD

**Discussion by ProonFace** [localsend/localsend#2118] [Discussion (English, International)]
When I open localsend, I cannot connect to the internet in localsend because of a socket error. "OS error: Only one use of each socket is permitted"
I tried deleting and reinstalling the app and resetting my internet. Any ways to fix this?

**Comment by Tienisto** [comment]
Have tried an older version of LocalSend? What OS do you use?

**Reply by ProonFace** [reply]
I have not tried an older version, I am on windows 11

**Reply by ProonFace** [reply]
It seems the issue is with port 53317 being allocated to my firewall... Should I disable this port or is there a way to change the port?

**Reply by Tienisto** [reply]
Yes, you can change the port in the advanced settings. But this needs to be done on all your devices.

**Reply by ProonFace** [reply]
Thanks for the help.  I killed 53317 in task manager which has the app working now, is there any risk to this to my firewall or way to reestablish this firewall port? Also, what do you recommend changing the port to?

**Reply by Tienisto** [reply]
Is your network configured as a "public" network? Usually, setting the network to "private" is enough and causes fewer problems. But I don't know if this is the cause.
I don't know what the firewall is doing on that port honestly...
The port number is not important, just use any port number between 1024 - 65535 (somewhere in the middle should be fine 10000~30000).

**Reply by natesiy** [reply]
Had to change it to the Default Port+1 to get it to work. All over devices that you want to send from also needs to be changed to the same port, or LocalSend won't detect the other devices.

**Comment by symonxdd** [comment]
In my case it was GitHub Desktop using that port. Upon closing the app, and restarting LocalSend / the server it started working.
Who would've thought...

**Reply by naufalist** [reply]
This really helps me, thank you

**Comment by TheEnbyWitch** [comment]
I'm getting the same problem but nothing else other than LocalSend is using the port. Changing the port to another unused one isn't doing anything either and I have allowed LocalSend through a firewall.

<img width="1236" height="706" alt="image" src="https://github.com/user-attachments/assets/6334265f-c067-4405-907c-e722f0993666" />

**Reply by nekoprog** [reply]
@TheEnbyWitch you can try this workaround https://github.com/localsend/localsend/issues/125#issuecomment-2533358231

**Reply by PikachuEXE** [reply]
The workaround above works for me

**Reply by ctkqiang** [reply]
You can try this for windows: 
1. Launch Powershell as Admin
2, Run these:
```poweshell
net stop winnat; net start winnat
```
3. Look for any port conflict (expect return NOTHING)
```powershell
netstat -ano | findstr :53317
```

**Comment by ctkqiang** [comment]
For Windows Users can try this,

1. Launch Powershell as Admin
2, Run these:
```poweshell
net stop winnat; net start winnat
```
3. Look for any port conflict (expect return NOTHING)
```powershell
netstat -ano | findstr :53317
```

--- 

For Mac Users,
1. Search For PORT conflict 
```zsh 
lsof -i :53317
```
2. Kill the Specific Process from the response earlier
```zsh
sudo kill -9 <PID>
```
